### ISSUE TITLE
Error in generation allocation

### ISSUE THREAD

**Issue by arengel** [catalyst-cooperative/pudl#2049]
### Describe the bug
An `AttributeError` occurs in `allocate_gen_fuel_by_generator_energy_source`.

### Bug Severity
How badly is this bug affecting you?
 - **High:** This bug is preventing me from using PUDL.

### To Reproduce
```python 
pt = pudl.output.pudltabl.PudlTabl(
    sqlalchemy.create_engine(pudl.workspace.setup.get_defaults()["pudl_db"]),
    freq="AS",
    fill_fuel_cost=True,
    roll_fuel_cost=True,
    fill_net_gen=True,
)
df = pt.gen_fuel_by_generator_energy_source_eia923()
```
When running this code I get the following error:
```
Traceback (most recent call last):
  File "/Users/aengel/mambaforge/envs/hub/lib/python3.10/site-packages/IPython/core/interactiveshell.py", line 3433, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-4-d690f2444caf>", line 1, in <module>
    df = pt.gen_fuel_by_generator_energy_source_eia923()
  File "/Users/aengel/mambaforge/envs/hub/lib/python3.10/site-packages/pudl/output/pudltabl.py", line 787, in gen_fuel_by_generator_energy_source_eia923
    ] = allocate_gen_fuel_by_generator_energy_source(pudl_out=self)
  File "/Users/aengel/mambaforge/envs/hub/lib/python3.10/site-packages/pudl/analysis/allocate_net_gen.py", line 271, in allocate_gen_fuel_by_generator_energy_source
    _ = test_original_gf_vs_the_allocated_by_gens_gf(
  File "/Users/aengel/mambaforge/envs/hub/lib/python3.10/site-packages/pudl/analysis/allocate_net_gen.py", line 1886, in test_original_gf_vs_the_allocated_by_gens_gf
    max_diff = gf_test[f"{data_col}_diff"].max().round(2)
AttributeError: 'float' object has no attribute 'round'
```

### Software Environment?
 - Operating System. macOS 12.6.1
 - Python version and distribution 3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:41:22) [Clang 13.0.1 ]
 - How did you install PUDL?
    - using pip from GitHub (`catalystcoop.pudl @ git+https://github.com/catalyst-cooperative/pudl.git@dev`) giving `0.6.1.dev1034+g50bbc573`.
    - I am using the pudl.sqlite that I downloaded today, running a sha256 checksum on it gives `e7975672303d8ad3a423cf80e9d15273b6e11171c40182f76dd6bda1f53272d4`.

**Comment by zaneselvans** [comment]
Hmm. I'm not able to reproduce this error, also on python 3.10.6, having just downloaded the PUDL DB from yesterday with the same sha256 sum. I wonder what's different.

**Comment by zaneselvans** [comment]
Fixed by PR #2052
